ruby-challenges.rb

```diff
@@ -1,32 +1,105 @@
 # ASSESSMENT 5: Ruby Coding Practical Questions
-# MINASWAN
+# MINASWAN = "This is the way..." - Paz Vizla (RIP)
 
 # --------------------1) Create a method that takes in an array of words and a single letter and returns an array of all the words containing that particular letter. Use the test variables provided.
 
+# Pseudo code
+    # Use the select method in Ruby, which returns a new array containing elements that satisfy a given condition.
+    # Pass a block that checks if each word in the words array includes the given letter, case-insensitive.
+    # Use the downcase method to ensure that the comparison is case-insensitive.
+
 beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
 
 letter_o = 'o'
 # Expected output: ['coffee', 'soda water']
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def words_containing_letter(words, letter)
+    filtered_words = words.select { |word| word.downcase.include?(letter.downcase) }
+    return filtered_words
+end
+
+p words_containing_letter(beverages_array, letter_o) # Output: ['coffee', 'soda water']
+
+p words_containing_letter(beverages_array, letter_t) # Output: ['tea', 'water', 'soda water']
+
+
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
+# Pseudo code:
+    # Flatten the nested arrays and get all values
+    # Sort the values in alphabetical order
+
+us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], northeast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+def ordered_states_unleashed(hash)
+    values = hash.values.flatten
+    values.sort!
+    return values
+end
+
```

Comment:
Same as before, remember implicit return. Also, you can chain the `flatten` and `sort` methods